irc/client.go

```diff
@@ -530,21 +538,10 @@ func (client *Client) destroy() {
 		client.server.connectionLimitsMutex.Unlock()
 	}
 
-	// remove from opers list
-	_, exists := client.server.currentOpers[client]
-	if exists {
-		delete(client.server.currentOpers, client)
-	}
-
 	// alert monitors
-	client.server.monitoringMutex.RLock()
-	for _, mClient := range client.server.monitoring[client.nickCasefolded] {
-		mClient.Send(nil, client.server.name, RPL_MONOFFLINE, mClient.nick, client.nick)
-	}
-	client.server.monitoringMutex.RUnlock()
-
-	// remove my monitors
-	client.clearMonitorList()
+	client.server.monitorManager.alertMonitors(client, false)
+	// clean up monitor state
+	client.server.monitorManager.clearMonitorList(client)
```

Comment:
hmm, honestly I'm iffy on the name of this function. Maybe it should be named something along the lines of... `removeClient` or something, to make clear it's removing this client from monitor lists and also deleting this client's monitor list?

Reply:
Oh wait a minute... yeah I think I can see this, silly me. I'm still a bit iffy on the function name but that one I suggested wouldn't fit, cleaning up the fn name and the comment on that function so it's more clear would be really good.